R/r_import.R

```diff
@@ -179,35 +152,13 @@ DataImport <- R6::R6Class(
     #' @param allow_dirty_git If TRUE then skips check that the import is up to
     #' date with remote git repo. FALSE by default.
     load = function(comment = NULL, dry_run = FALSE, allow_dirty_git = FALSE) {
-      if (is.null(private$transformed_data)) {
-        stop("Cannot run load as no data has been transformed.")
-      }
-      if (!private$transform_passed) {
-        stop("Cannot run load as transform tests failed.")
-      }
-      if (!is.null(private$require_branch)) {
-        if (git_branch(self$path) != private$require_branch) {
-          stop(sprintf("This import can only be run from the '%s' branch",
-                       private$require_branch), call. = FALSE)
-        }
-      }
-      if (private$confirm) {
-        confirmed <- askYesNo(
-          sprintf(
-            "About to upload to database '%s' are you sure you want to proceed?",
-            private$db_name),
-          default = FALSE)
-        if (is.na(confirmed) || !confirmed) {
-          message("Not uploading to database.")
-          return(invisible(FALSE))
-        }
+      checks_passed <- self$pre_modify_checks(dry_run, allow_dirty_git)
+      if (isFALSE(checks_passed)) {
+        return(invisible(FALSE))
       }
       message(sprintf("Running load %s", self$path))
-      if (!allow_dirty_git && !dry_run && !git_repo_is_clean(self$path)) {
-        stop("Can't run load as repository has unstaged changes. Update git or run in dry-run mode.")
-      }
 
-      use_transaction <- private$load_in_transaction || dry_run
+      use_transaction <- private$modify_in_transaction || dry_run
       if (use_transaction) {
         self$begin_transaction()
```

Comment:
I was also thinking about adding a function on the super class called like `with_transaction` or something which would take a block of code and manage the rollback or commit depending on outcome but thought being explicit here in the function might be more clear for reading.